### PR TITLE
[bitnami/prometheus-operator] Use requests and limits for thanos sidecar

### PR DESCRIPTION
**Pull request by hwoarang**
Add the ability to configure requests and limits for Thanos sidecar so
we have better control over the entire prometheus pods.

* Allows us to fine tune pod usage to improve k8s scheduling

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/prometheus-operator/values-production.yaml [hunk 1]
```diff
@@ -683,6 +683,11 @@ prometheus:
     #   secretName: thanos-objstore-config
     #   secretKey: thanos.yaml
 
+    ## Configure resource requests and limits
+    ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+    ##
+    resources: {}
```
Comment (juan131) [inline bitnami/prometheus-operator/values-production.yaml:689]: To be consistent with other charts, please use:
```suggestion
    ## Thanos sidecar container resource requests and limits.
    ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
    ##
    resources:
      ## We usually recommend not to specify default resources and to leave this as a conscious
      ## choice for the user. This also increases chances charts run on environments with little
      ## resources, such as Minikube. If you do want to specify resources, uncomment the following
      ## lines, adjust them as necessary, and remove the curly braces after 'resources:'.
      limits: {}
      #   cpu: 100m
      #   memory: 128Mi
      requests: {}
      #   cpu: 100m
      #   memory: 128Mi
```

File: bitnami/prometheus-operator/values.yaml [hunk 2]
```diff
@@ -689,6 +689,11 @@ prometheus:
     #   secretName: thanos-objstore-config
     #   secretKey: thanos.yaml
 
+    ## Configure resource requests and limits
+    ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+    ##
+    resources: {}
+
```
Comment (juan131) [inline bitnami/prometheus-operator/values.yaml:696]: Same here:
```suggestion
    ## Thanos sidecar container resource requests and limits.
    ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
    ##
    resources:
      ## We usually recommend not to specify default resources and to leave this as a conscious
      ## choice for the user. This also increases chances charts run on environments with little
      ## resources, such as Minikube. If you do want to specify resources, uncomment the following
      ## lines, adjust them as necessary, and remove the curly braces after 'resources:'.
      limits: {}
      #   cpu: 100m
      #   memory: 128Mi
      requests: {}
      #   cpu: 100m
      #   memory: 128Mi
```

File: bitnami/prometheus-operator/templates/prometheus/prometheus.yaml [hunk 3]
```diff
@@ -207,6 +207,7 @@ spec:
               name: {{ .Values.prometheus.thanos.objectStorageConfig.secretName }}
               key: {{ .Values.prometheus.thanos.objectStorageConfig.secretKey | default "thanos.yaml" }}
       {{- end }}
+      resources: {{- toYaml .Values.prometheus.thanos.resources | nindent 8 }}
```
Comment (juan131) [inline bitnami/prometheus-operator/templates/prometheus/prometheus.yaml:210]: Avoid setting it when the parameter is not set:
```suggestion
      {{- if .Values.prometheus.thanos.resources }}
      resources: {{- toYaml .Values.prometheus.thanos.resources | nindent 8 }}
      {{- end }}
```

File: bitnami/prometheus-operator/README.md [hunk 4]
```diff
@@ -248,6 +248,7 @@ The following table lists the configurable parameters of the Prometheus Operator
...
 | `prometheus.thanos.service.extraPorts`                | Additional ports to expose from the Thanos sidecar container                                                                   | `[]`                                                                                                                                                                                                                                                |
+| `prometheus.thanos.resources`                         | CPU/Memory resource requests/limits for the Thanos sidecar                                                                     | `{}`                                                                                                                                                                                                                                                |
```
Comment (juan131) [inline bitnami/prometheus-operator/README.md:251]: To be consistent with other charts:
```suggestion
| `prometheus.thanos.resources.limits`                  | The resources limits for the Thanos sidecar container                                      | `{}`                                                    |
| `prometheus.thanos.resources.requests`                | The requested resources for the Thanos sidecar container                                   | `{}`                                                    |
```

File: bitnami/prometheus-operator/values-production.yaml (hunk 1 above)
Comment (hwoarang) [reply bitnami/prometheus-operator/values-production.yaml:689]: done

File: bitnami/prometheus-operator/values.yaml (hunk 2 above)
Comment (hwoarang) [reply bitnami/prometheus-operator/values.yaml:696]: done

File: bitnami/prometheus-operator/templates/prometheus/prometheus.yaml (hunk 3 above)
Comment (hwoarang) [reply bitnami/prometheus-operator/templates/prometheus/prometheus.yaml:210]: done

File: bitnami/prometheus-operator/README.md (hunk 4 above)
Comment (hwoarang) [reply bitnami/prometheus-operator/README.md:251]: done

**Comment by hwoarang** [comment]
@juan131 thank you for your suggestions. I have addressed all your comments.

**Pull request merged**